client/cli/src/lib.rs

```diff
@@ -161,637 +127,200 @@ fn base_path(
 		)
 }
 
-/// Check whether a node name is considered as valid
-fn is_node_name_valid(_name: &str) -> Result<(), &str> {
-	let name = _name.to_string();
-	if name.chars().count() >= NODE_NAME_MAX_LENGTH {
-		return Err("Node name too long");
-	}
-
-	let invalid_chars = r"[\\.@]";
-	let re = Regex::new(invalid_chars).unwrap();
-	if re.is_match(&name) {
-		return Err("Node name should not contain invalid chars such as '.' and '@'");
-	}
-
-	let invalid_patterns = r"(https?:\\/+)?(www)+";
-	let re = Regex::new(invalid_patterns).unwrap();
-	if re.is_match(&name) {
-		return Err("Node name should not contain urls");
-	}
-
-	Ok(())
+/// Gets the struct from the command line arguments.  Print the
+/// error message and quit the program in case of failure.
+pub fn from_args<T>(version: &VersionInfo) -> T
+where
+	T: StructOpt + Sized,
+{
+	from_iter::<T, _>(&mut std::env::args_os(), version)
 }
 
-/// Parse command line interface arguments and prepares the command for execution.
-///
-/// Before returning, this function performs various initializations, such as initializing the
-/// panic handler and the logger, or increasing the limit for file descriptors.
-///
-/// # Remarks
-///
-/// `CC` is a custom subcommand. This needs to be an `enum`! If no custom subcommand is required,
-/// `NoCustom` can be used as type here.
-///
-/// `RP` are custom parameters for the run command. This needs to be a `struct`! The custom
-/// parameters are visible to the user as if they were normal run command parameters. If no custom
-/// parameters are required, `NoCustom` can be used as type here.
-pub fn parse_and_prepare<'a, CC, RP, I>(
-	version: &'a VersionInfo,
-	impl_name: &'static str,
-	args: I,
-) -> ParseAndPrepare<'a, CC, RP>
+/// Gets the struct from any iterator such as a `Vec` of your making.
```

Comment:
Needs better documentation. It doesn't even mention that it's for parsing command-line arguments.

Reply:
Will improve that. I copy-pasted it from structopt's doc